packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,940 @@
+/**
+ * External dependencies
+ */
+// This is required to fix the "regeneratorRuntime is not defined" error
+import '@automattic/calypso-polyfills';
+
+/**
+ * Internal dependencies
+ */
+import createExPlatClient from '../create-explat-client';
+import { delayedValue, validExperimentAssignment } from '../internal/test-common';
+import * as Timing from '../internal/timing';
+import type { Config, ExperimentAssignment } from '../types';
+
+const spiedMonotonicNow = jest.spyOn( Timing, 'monotonicNow' );
+
+const mockedFetchExperimentAssignment = jest.fn();
+const mockedGetAnonId = jest.fn();
+const mockedLogError = jest.fn();
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore; Not using the full Config
+const mockedConfig: Config = {
```

Comment:
Ah I did this to make `mockFetchExperimentAssignmentToMatchExperimentAssignment` easier to use but fair enough, not too hard to make it work.

Reply:
Yeah, I was thinking we could also make the overrides more sophisticated so that passing an `ExperimentAssignment` object as the value of `fetchExperimentAssignment` would also create the mock, but I'm not sure it'd be that clean without playing with it.

Reply:
So I have added `createConfigMock` so far but now I am thinking that this doesn't add much - mocks are reset anyway in-between tests, and making this work well with typescript is a bit of a pain as it needs to fit the two different uses - as a Config and as a mocked Config (e.g. to access `.mockImplement()`).